addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/RequestorJob.java

```diff
@@ -0,0 +1,221 @@
+                                                responseCode,
+                                                request));
+                            }
+                        } catch (NumberFormatException e) {
+                            progress.warn(
+                                    Constant.messages.getString(
+                                            "automation.error.requestor.badcode",
+                                            this.getName(),
+                                            responseCode,
+                                            request));
+                        }
+                    }
+                }
+            }
+        } else {
+            progress.error(Constant.messages.getString("automation.error.requestor.badlist", o));
+        }
+    }
+
+    @Override
+    public void runJob(
+            AutomationEnvironment env, LinkedHashMap<?, ?> jobData, AutomationProgress progress) {
+        ArrayList<?> requests = (ArrayList<?>) jobData.get("requests");
+        if (requests != null) {
+            for (Object request : requests) {
+                LinkedHashMap<?, ?> requestMap = (LinkedHashMap<?, ?>) request;
+                HttpMessage msg = new HttpMessage();
+                msg.getRequestHeader().setMethod((String) requestMap.get("method"));
+                try {
+                    msg.getRequestHeader().setURI(new URI((String) requestMap.get("url"), true));
+                } catch (URIException e) {
+                    // Handled above
+                }
+                String name;
+                if (requestMap.get("name") != null) {
+                    name = (String) requestMap.get("name");
+                } else {
+                    name =
+                            msg.getRequestHeader().getMethod()
+                                    + msg.getRequestHeader().getURI().toString();
+                }
+                if (requestMap.get("data") != null) {
+                    msg.getRequestBody().setBody((String) requestMap.get("data"));
+                    msg.getRequestHeader().setContentLength(msg.getRequestBody().length());
+                }
+                try {
+                    httpSender.sendAndReceive(msg);
+                } catch (IOException e) {
+                    progress.warn(
+                            Constant.messages.getString(
+                                    "automation.error.requestor.badnetwork",
+                                    this.getName(),
+                                    name,
+                                    e));
+                }
+                persistToHistoryAndSitesTree(msg);
+                if (requestMap.containsKey("responseCode")) {
+                    Object o = requestMap.get("responseCode");
+                    if (o instanceof Integer) {
+                        int expectedCode = (Integer) o;
+                        int receivedCode = msg.getResponseHeader().getStatusCode();
```

Comment:
This code will not be correct if the response was not received, should probably skip this check altogether.